core/src/test/scala/org/apache/spark/crypto/JceAesCtrCryptoCodecSuite.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.crypto
+
+import java.io.{ByteArrayInputStream, BufferedOutputStream, ByteArrayOutputStream}
+import java.security.SecureRandom
+
+import org.apache.spark.crypto.CommonConfigurationKeys._
+import org.apache.spark.{SparkFunSuite, SparkConf, Logging}
+
+/**
+ * test JceAesCtrCryptoCodec
+ */
+class JceAesCtrCryptoCodecSuite extends SparkFunSuite with Logging {
+
+  test("TestCryptoCodecSuite") {
+    val random = new SecureRandom
+    val dataLen = 10000000
+    val inputData = new Array[Byte](dataLen)
+    val outputData = new Array[Byte](dataLen)
+    random.nextBytes(inputData)
+    // encrypt
+    val sparkConf = new SparkConf
+    sparkConf.set(SPARK_SECURITY_CRYPTO_CODEC_CLASSES_AES_CTR_NOPADDING_KEY,
+      classOf[JceAesCtrCryptoCodec].getName)
+    val cipherSuite = new CipherSuite("AES/CTR/NoPadding", 16)
+    val codec = new JceAesCtrCryptoCodec(sparkConf)
+    val aos = new ByteArrayOutputStream
+    val bos = new BufferedOutputStream(aos)
+    val key = new Array[Byte](16)
+    val iv = new Array[Byte](16)
+    random.nextBytes(key)
+    random.nextBytes(iv)
+
+    val cos = new CryptoOutputStream(bos, codec, 1024, key, iv)
+    cos.write(inputData, 0, inputData.length)
```

Comment:
It would be nice if the test exercised more code paths; right now it's writing a huge byte array. It should also try to write small byte arrays that are smaller than the configured buffer size, for example. It should also make that the data leaves an unfinished buffer at the end, to make sure that the `flush` implementation is doing the right thing.